### PR TITLE
Fix invalid exceptions in `parseDateTime` with `%F` and `%D` placeholders

### PR DESCRIPTION
**Pull request by antonio2368**
### Changelog category (leave one):
- Bug Fix (user-visible misbehavior in an official stable release)

### Changelog entry (a user-readable short description of the changes that goes to CHANGELOG.md):
Fix invalid exceptions in function `parseDateTime` with `%F` and `%D` placeholders.

### Documentation entry for user-facing changes

- [ ] Documentation is written (mandatory for new features)

> Information about CI checks: https://clickhouse.com/docs/en/development/continuous-integration/

#### CI Settings (Only check the boxes if you know what you are doing):
- [ ]  Allow: All Required Checks
- [ ]  Allow: Stateless tests
- [ ]  Allow: Stateful tests
- [ ]  Allow: Integration Tests
- [ ]  Allow: Performance tests
- [ ]  Allow: All Builds
- [ ]  Allow: batch 1, 2 for multi-batch jobs
- [ ]  Allow: batch 3, 4, 5, 6 for multi-batch jobs
---
- [ ]  Exclude: Style check
- [ ]  Exclude: Fast test
- [ ]  Exclude: All with ASAN
- [ ]  Exclude: All with TSAN, MSAN, UBSAN, Coverage
- [ ]  Exclude: All with aarch64, release, debug
---
- [ ]  Do not test
- [ ]  Woolen Wolfdog
- [ ]  Upload binaries for special builds
- [ ]  Disable merge-commit
- [ ]  Disable CI cache

### CODE REVIEW

**Comment by robot-ch-test-poll** [comment]
*This is an automated comment for commit 988ba2e22dba550d5610a3658571d0bb3493b4c9 with description of existing statuses. It's updated for the latest CI running*

[✅ Click here](https://s3.amazonaws.com/clickhouse-test-reports/65768/988ba2e22dba550d5610a3658571d0bb3493b4c9/ci_running.html) to open a full report in a separate page

<details><summary>Successful checks</summary>
<table>
<thead><tr><th>Check name</th><th>Description</th><th>Status</th></tr></thead>
<tbody>
<tr><td>Docs check</td><td>Builds and tests the documentation</td><td>✅ success</td></tr>
<tbody>
</table>
</details>

File: tests/queries/0_stateless/03197_fix_parse_mysql_iso_date.sql [hunk 1]
```diff
@@ -0,0 +1 @@
+SELECT parseDateTime(formatDateTime(toDateTime('2024-06-20 12:00:00'), '%F %T', 'Europe/Paris'), '%F %T', 'UTC') AS x
```
Comment (rschu1ze) [inline tests/queries/0_stateless/03197_fix_parse_mysql_iso_date.sql:1]: If I may wish for one thing: Please put this at the end of tests/queries/0_stateless/02668_parse_datetime.sql (which already contains a few tests for crashes), so we keep related tests together.

Comment (antonio2368) [reply tests/queries/0_stateless/03197_fix_parse_mysql_iso_date.sql:1]: AFAIK we prefer creating new tests so we have a better history of its failure

Comment (antonio2368) [reply tests/queries/0_stateless/03197_fix_parse_mysql_iso_date.sql:1]: But I'm fine either way

**Review by rschu1ze — APPROVED** [review]

**Comment by clickgapai** [comment]
Hi @antonio2368 @rschu1ze — while reviewing this PR I found the following:

- **Bug (P1):** `mysqlAmericanDate` uses unconditional `+2000` for year instead of proper 2-digit year conversion — https://github.com/ClickHouse/ClickHouse/issues/104895

Happy to discuss — close anything that's wrong or already addressed.

**Pull request merged**